### ISSUE TITLE
imapd.conf:cyrus_user ignored by ctl_cyrusdb

### ISSUE THREAD

**Issue by dilyanpalauzov** [cyrusimap/cyrus-imapd#2872]
imapd.conf:cyrus_user specifies the user, under which the cyrus imap processes shall run as.  The user is set at runtime by lib/util.c:become_cyrus(), which calls: lib/util.c:cyrus_user():
```c
EXPORTED const char *cyrus_user(void)
{
    const char *cyrus = getenv("CYRUS_USER");
    if (!cyrus) cyrus = config_getstring(IMAPOPT_CYRUS_USER);
    if (!cyrus) cyrus = CYRUS_USER;
    assert(cyrus != NULL);
    return cyrus;
}
```

ctl_cyrusdb:main() calls first `if ((geteuid()) == 0 && (become_cyrus(/*is_master*/0) != 0))` and then  `cyrus_init(alt_config, "ctl_cyrusdb", 0, 0);` and imap/global.c:cyrus_init() calls `config_read(alt_config, config_need_data);`.

When ctl_cyrusdb is run under strace, it shows that first `setuid("cyrus")` is called and then the config file is read.  Thus ctl_cyrusdb ignores imapd.conf:cyrus_user.  Effectively the value of imapd.conf:cyrus_user is ignored by ctl_cyrusdb.

**Comment by elliefm** [comment]
I guess this is for 3.0?  Looks like this exact thing was fixed on master by 0f7de0a7f7, I'll see if it backports...

**Comment by elliefm** [comment]
cherry-picked (with some handholding) to 3.0 as c230030a5
